internal/core/src/index/InvertedIndexTantivy.cpp

```diff
@@ -447,6 +331,176 @@ InvertedIndexTantivy<T>::BuildWithRawData(size_t n,
+            switch (schema_.element_type()) {
+                case proto::schema::DataType::Bool: {
+                    build_index_for_array<bool>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Int8: {
+                    build_index_for_array<int8_t>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Int16: {
+                    build_index_for_array<int16_t>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Int32: {
+                    build_index_for_array<int32_t>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Int64: {
+                    build_index_for_array<int64_t>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Float: {
+                    build_index_for_array<float>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::Double: {
+                    build_index_for_array<double>(field_datas);
+                    break;
+                }
+
+                case proto::schema::DataType::VarChar: {
+                    for (const auto& data : field_datas) {
+                        auto n = data->get_num_rows();
+                        auto array_column =
+                            static_cast<const Array*>(data->Data());
+                        for (int64_t i = 0; i < n; i++) {
+                            assert(
+                                array_column[i].get_element_type() ==
+                                static_cast<DataType>(schema_.element_type()));
+                            std::vector<std::string> output;
+                            for (int64_t j = 0; j < array_column[i].length();
+                                 j++) {
+                                output.push_back(
+                                    array_column[i]
+                                        .template get_data<std::string>(j));
+                            }
+                            wrapper_->template add_multi_data(output.data(),
+                                                              output.size());
+                        }
+                    }
+                    break;
+                }
+
+                default:
+                    PanicInfo(ErrorCode::NotImplemented,
+                              fmt::format("Inverted index not supported on {}",
+                                          schema_.element_type()));
+            }
+            break;
+        }
+
+        default:
+            PanicInfo(ErrorCode::NotImplemented,
+                      fmt::format("Inverted index not supported on {}",
+                                  schema_.data_type()));
+    }
+}
+
+template <typename T>
+template <typename ElementT>
```

Comment:
for your framework, T and ElementT is same, is that right?  if same, not need ElementT

Reply:
T and ElementT is not same for the array data type in the first design.

Reply:
And yes they are same now.